transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystemRootResourceDefinition.java

```diff
@@ -521,12 +521,13 @@ protected boolean applyUpdateToRuntime(final OperationContext context, final Mod
             throws OperationFailedException {
             int timeout = resolvedValue.asInt();
 
-            TxControl.setDefaultTimeout(timeout);
             if (timeout == 0) {
                 ModelNode model = context.readResource(PathAddress.EMPTY_ADDRESS).getModel();
                 timeout = MAXIMUM_TIMEOUT.resolveModelAttribute(context, model).asInt();
                 TransactionLogger.ROOT_LOGGER.timeoutValueIsSetToMaximum(timeout);
             }
+            arjPropertyManager.getCoordinatorEnvironmentBean().setDefaultTimeout(timeout);
```

Comment:
These lines should be moved to before checking the timeout equal 0, as TxControl allow 0 with the timeout which means no timeout at all.

Reply:
ah, correct. thank you!